test/test_external.py

```diff
@@ -417,5 +418,20 @@ def test_respect_queue_when_load_from_config():
             assert interface("bob") == "foo"
 
 
+def test_raise_value_error_when_api_name_invalid():
+    with pytest.raises(InvalidApiName):
+        demo = gr.Blocks.load(name="spaces/gradio/hello_world")
+        demo("freddy", api_name="route does not exist")
+
+
+def test_use_api_name_in_call_method():
+    # Interface
+    demo = gr.Blocks.load(name="spaces/gradio/hello_world")
+    assert isinstance(demo("freddy", api_name="predict"), str)
```

Comment:
Suggestion -- instead of just checking the type, can we assert the correct value for a more stringent test?

Reply:
For sure!